### PR TITLE
chore(manifest): add sidecar config to manifest and make more fields overridable

### LINKED ISSUE #949: Update manifest pkg to parse config for explicit sidecar
To support explicit sidecar in our CLI, we need to update `manifest` pkg so that any configuration for explicit sidecar in manifest can be properly parsed.

### PR DESCRIPTION
**Pull request by iamhopaul123**
fix #949. And make fields except for service name overridable in the manifest. Previously `image` and `logRetention` cannot be override.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/manifest/svc.go [hunk 1]
```diff
@@ -41,6 +41,47 @@ type ServiceImageWithPort struct {
...
+// Sidecar holds configuration for all sidecar containers in a service.
+type Sidecar struct {
+	Sidecars map[string]SidecarConfig `yaml:"sidecars"`
+}
+
+// SidecarConfig represents the configurable options for setting up a sidecar container.
+type SidecarConfig struct {
+	Port      string `yaml:"port"`
+	Image     string `yaml:"image"`
+	CredParam string `yaml:"credentialParameter"`
```
Comment (efekarakus) [inline internal/pkg/manifest/svc.go:53]:
```suggestion
	CredsParam string `yaml:"credentialsParameter"`
```
to be consistent with https://docs.aws.amazon.com/AmazonECS/latest/developerguide/private-auth.html

File: internal/pkg/manifest/svc.go [hunk 2]
```diff
@@ -41,6 +41,47 @@ type ServiceImageWithPort struct {
...
+}
+
+func (tc Sidecar) copyAndApply(other Sidecar) Sidecar {
```
Comment (efekarakus) [inline internal/pkg/manifest/svc.go:56]: nit: should  the receiver be just `s`?

Comment (iamhopaul123) [reply internal/pkg/manifest/svc.go:56]: oh yeah good catch! I always forget this

File: internal/pkg/manifest/svc.go (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/manifest/svc.go:53]: `credentialsParameter` is consistent with the task def. However, in test def it is nested in `repositoryCredentials`, which I feel in Copilot we could prob simplify this unless in the future they add additional params under `repositoryCredentials`. What do you think?

Comment (iamhopaul123) [reply internal/pkg/manifest/svc.go:53]: nvm i'll update to `credentialsParameter`

**Pull request merged**